sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -776,6 +786,63 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
         ///
         internal virtual PartitionManager CreateStorageManager(BlobContainerClient checkpointStore) => new BlobsCheckpointStore(checkpointStore, RetryPolicy);
 
+        /// <summary>
+        ///   Starts running a task responsible for receiving and processing events in the context of a specified partition.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition the task is associated with.  Events will be read only from this partition.</param>
+        /// <param name="startingPosition">The position within the partition where the task should begin reading events.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The running task that is currently receiving and processing events in the context of the specified partition.</returns>
+        ///
+        internal virtual Task RunPartitionProcessingAsync(string partitionId,
+                                                          EventPosition startingPosition,
+                                                          CancellationToken cancellationToken) => Task.Run(async () =>
+        {
+            var emptyPartitionContext = new EmptyPartitionContext(partitionId);
+
+            await using var connection = ConnectionFactory();
+            await using var consumer = CreateConsumer(ConsumerGroup, connection, ProcessingConsumerOptions);
+            await foreach (var partitionEvent in consumer.ReadEventsFromPartitionAsync(partitionId, startingPosition, ProcessingReadEventOptions, cancellationToken))
```

Comment:
Orthogonal to your change, but do you mind adding `ConfigureAwait(false)` on this?